apps/spruce/src/pages/waterfall/index.tsx

```diff
@@ -31,6 +42,19 @@ const Waterfall: React.FC = () => {
           </Banner>
         )}
         <WaterfallFilters projectIdentifier={projectIdentifier ?? ""} />
+        <BadgeWrapper>
+          <FilterBadges
+            badges={badges}
+            onClearAll={() => {
+              sendEvent({ name: "Deleted all filter badges" });
+              handleClearAll();
+            }}
+            onRemove={(b) => {
+              sendEvent({ name: "Deleted one filter badge" });
+              handleOnRemove(b);
+            }}
```

Comment:
Since we're using Suspense on this page, these handlers represent a perfect use case for [`useTransition`](https://react.dev/reference/react/useTransition). Notice how after "clear all" is clicked on a particularly heavy page, the other page elements become unusable. Adding `startTransition` to these handlers fixes that issue so the page is responsive in the middle of a rerender.